provider/resources.go

```diff
@@ -626,6 +615,30 @@ func preConfigureCallback(vars resource.PropertyMap, c shim.ResourceConfig) erro
 	return nil
 }
 
+// We should only run the validation once to avoid duplicating the reported errors.
+var credentialsValidationCounter atomic.Int32
```

Comment:
Nit: We have `atomic.Bool` with `if !credentialsValidated.CompareAndSwap(false, true) { return nil }`

We also have `sync.OnceFunc` that exists to do exactly what you have implemented in atomics.

Neither of these options are vulnerable to overflow (though I doubt that will be a problem).

Reply:
sync.Once is nice actually, I forgot that one.

Reply:
Great suggestion, thank you! I think `OnceFunc` would be a bit tricky to use here but `sync.Once` worked well and made it clearer.